src/ott/initializers/linear/initializers.py

```diff
@@ -313,3 +338,80 @@ def body_fn(i: int, f: jnp.ndarray) -> jnp.ndarray:
     return f.at[i].set(new_f)
 
   return jax.lax.fori_loop(0, len(f), body_fn, f)
+
+
+@jax.tree_util.register_pytree_node_class
+class SubsampleInitializer(DefaultInitializer):
+  """Subsample initializer :cite:`thornton2022rethinking:22`.
+
+  Subsample each :class:~ott.geometry.point_cloud.PointCloud, then compute closed from
+  Sinkhorn potential between subsampled approximations. Use this potential
+  to initialize Sinkhorn potentials/scalings.
+
+  Args:
+    subsample_n: number of points to subsample from each point cloud.
+  """
+
+  def __init__(
+      self,
+      subsample_n: int,
+      subsample_n_y: Optional[int] = None,
+      sinkhorn_kwargs: Optional[Dict[str, Any]] = None,
+  ):
+    super().__init__()
+    self.subsample_n = subsample_n
+    self.subsample_n_y = subsample_n_y or subsample_n
+    self.sinkhorn_kwargs = sinkhorn_kwargs or {}
+
+  def init_dual_a(
+      self, ot_prob: linear_problem.LinearProblem, lse_mode: bool,
+      rng: jax.random.PRNGKey
+  ) -> jnp.ndarray:
+    """Subsample initializer function.
+
+    Args:
+      ot_prob: OT problem between discrete distributions of size n and m.
+      lse_mode: Return potential if true, scaling if false.
+      rng: random number generator for subsampling.
+
+    Returns:
+      potential/scaling, array of size n.
+    """
+    from ott.solvers.linear import sinkhorn
+
+    assert isinstance(
+        ot_prob.geom, pointcloud.PointCloud
+    ), "Subsample initializer valid only for point clouds."
+
+    x, y = ot_prob.geom.x, ot_prob.geom.y
+    a, b = ot_prob.a, ot_prob.b
+
+    # subsample
+    rng_x, rng_y = jax.random.split(rng)
+    sub_x = jax.random.choice(
+        key=rng_x, a=x, shape=(self.subsample_n,), replace=True, p=a, axis=0
+    )
+    sub_y = jax.random.choice(
+        key=rng_y, a=y, shape=(self.subsample_n_y,), replace=True, p=b, axis=0
+    )
+
+    # create subsampled point cloud geometry
+    sub_geom = pointcloud.PointCloud(sub_x, sub_y, cost_fn=ot_prob.geom.cost_fn)
```

Comment:
Shouldn't also other arguments, such as `epsilon/scale_cost` be passed/copied from the original point cloud?

Reply:
Is there a tidy way of doing this in case arguments of point cloud change in future, to copy all attributed or inputs?

Reply:
Currently, no, will be addressed in #169 .